### PR TITLE
[CI] Fix build reason if and uses a better branch name.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

File: tools/devops/automation/templates/publish-html-result.yml [hunk 1]
```diff
@@ -30,14 +30,17 @@ steps:
     Write-Host "##vso[task.setvariable variable=BuildId;isOutput=true]$($config.BuildId)"
     Write-Host "##vso[task.setvariable variable=Commit;isOutput=true]$($config.Commit)"
     # calculate branch name and destination dir
-    $branchName = "$($config.BuildSourceBranchName)/$($config.Commit)/$($config.BuildId)"
     $prefix = ""
-    if ($Env:OriginalBuildReason -eq "PullRequest") {
+    $branchName = ""
+    if ($config.BuildReason -eq "PullRequest") {
       $prefix = "pr"
+      $branchName = "PR$(config.Commit)"
```
Comment (mandel-macaque) [inline tools/devops/automation/templates/publish-html-result.yml:37]:
```suggestion
      $branchName = "PR$($config.Commit)"
```

File: tools/devops/automation/templates/publish-html-result.yml [hunk 2]
```diff
@@ -30,14 +30,17 @@ steps:
...
+    if ($config.BuildReason -eq "PullRequest") {
       $prefix = "pr"
+      $branchName = "PR$($config.Commit)"
     } else {
       $prefix = "ci"
+      $branchName = $config.BuildSourceBranchName
     }
-    $branchName = "$prefix/$branchName"
+
+    $branchName = "$prefix/$branchName/$($config.Commit)/$($config.BuildId)"
```
Comment (spouliot) [inline tools/devops/automation/templates/publish-html-result.yml:43]: does not that duplicate `($config.Commit)` for PR ?

>  $branchName = "PR$($config.Commit)"
then
> $branchName = "$prefix/$branchName/$($config.Commit)/$($config.BuildId)"

Comment (mandel-macaque) [reply tools/devops/automation/templates/publish-html-result.yml:43]: @spouliot so the commit in PRs is the PR number, I can remove that

File: tools/devops/automation/templates/publish-html-result.yml [hunk 3]
```diff
@@ -30,14 +30,14 @@ steps:
     Write-Host "##vso[task.setvariable variable=BuildId;isOutput=true]$($config.BuildId)"
     Write-Host "##vso[task.setvariable variable=Commit;isOutput=true]$($config.Commit)"
     # calculate branch name and destination dir
-    $branchName = "$($config.BuildSourceBranchName)/$($config.Commit)/$($config.BuildId)"
-    $prefix = ""
-    if ($Env:OriginalBuildReason -eq "PullRequest") {
-      $prefix = "pr"
+    $branchName = ""
+    if ($config.BuildReason -eq "PullRequest") {
+      $branchName = "pr/PR$(config.Commit)"
```
Comment (mandel-macaque) [inline tools/devops/automation/templates/publish-html-result.yml:35]:
```suggestion
      $branchName = "pr/PR$($config.Commit)"
```

**Pull request merged**